Source/DafnyCore/Compilers/CSharp/Compiler-Csharp.cs

```diff
@@ -2881,9 +2881,27 @@ protected class ClassWriter : IClassWriter {
           }
 
         case BinaryExpr.ResolvedOpcode.LeftShift:
-          opString = "<<"; truncateResult = true; convertE1_to_int = true; break;
+          if (resultType.NormalizeExpand() is
+              BitvectorType { Width: var width and (32 or 64) }) {
```

Comment:
Is there a way to use `resultType.AsBitvectorType` and still use the `{ Width ... }` form? For example, is it legal to say

``` c#
if (resultType.AsBitvectorType is { Width: var width and (32 or 64) }) {
```

Reply:
Yes, let me do that.